app/api/util/collections/dict.py

```diff
@@ -0,0 +1,36 @@
+#
+# Dictionaries with additional behaviour.
+#
+
+import collections
+
+
+class LeastRecentlyUsedDict(collections.OrderedDict):
```

Comment:
should this file be named `least_recently_used_dict.py`?

Reply:
And in `app/api/util/collections/__init__.py` we can do:

```
from .least_recently_used_dict import LeastRecentlyUsedDict
```

And to use it:

```
from api.util.collections import LeastRecentlyUsedDict

LeastRecentlyUsedDict(maxsize=4)
```

Which I think is a little cleaner than:

```
from api.util.collections.dict import dict_util

dict_util.LeastRecentlyUsedDict(maxsize=4)
```

Reply:
Used the name dict as this could contain other dict related utilities in future. I added the import to `__init__.py`.